Add securitas direct integration

nragon opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add newly proposed [Securitas Direct](https://www.securitasdirect.pt/) home security integration. This integration allows users to manage their Securitas Direct installation. With this initial integration only the alarm panel will be added.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml entry
securitas_direct:
  username: USERNAME
  password: PASSWORD
  installation: INSTALLATION_NUMBER
  code: 1234
  country: PT
  lang: pt
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- Link to brands pull request: https://github.com/home-assistant/brands/pull/2107
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/15988

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

nragon committed:
Added securitas direct integration

nragon committed:
Added securitas direct integration (fixed requirements)

frenck commented on homeassistant/components/securitas_direct/__init__.py:
```diff
@@ -0,0 +1,114 @@
+MIN_SCAN_INTERVAL = timedelta(seconds=20)
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=40)
+
+CONFIG_SCHEMA = vol.Schema(
```
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

MartinHjelmare replied:
Please use our scaffold script to generate best practice code:
https://developers.home-assistant.io/docs/creating_component_index

frenck commented:
As a side question (as a Securitas Direct customer myself), we already have the `verisure` integration, how is this integration different?

nragon commented:
Securitas direct is part of verisure but they have different solutions.
For instance, mobile app is different, therefore, since we've made some reverse engineering, we found out that even the API is different.
You can check https://pypi.org/project/pysecuritas, which is the client we are using.

nragon committed:
Fixed arming/disarming state
Fixed code validation

nragon committed:
Reverted bad commit

nragon commented:
Any updates?

MartinHjelmare commented:
Sorry, we have a long queue of PRs to review. If you have time, please help out and review some PRs. :+1:

alnavasa commented:
> Sorry, we have a long queue of PRs to review. If you have time, please help out and review some PRs. 👍

45 days ago since this. Is there any issue with the code? I was expecting to see it implemented in todays update.

nragon commented:
> > Sorry, we have a long queue of PRs to review. If you have time, please help out and review some PRs. 👍
> 
> 45 days ago since this. Is there any issue with the code? I was expecting to see it implemented in todays update.

Actually, not complaining because, I also don't have the time to review PRs. But I think I saw some PRs merged that were developed after this (like mid/end of Jan 2021). Is there any specific criteria or priority on integrations?

MartinHjelmare commented:
Smaller PRs will be merged faster than larger PRs. Fixes will be merged faster than new features. New integrations usually needs to wait the longest since it's a new feature and often not very small PR.

There's not a strict FIFO queue in reviews. There are 300 plus PRs currently in the core repository, and that backlog creates problems in itself.

Please be patient. Thank you for your understanding.

Xeovar commented:
Any way of using this code while waiting for it getting released? This is the last piece missing in my HA installation...

MartinHjelmare commented on homeassistant/components/securitas_direct/__init__.py:
```diff
@@ -0,0 +1,135 @@
+
+async def async_setup(hass, config) -> bool:
+    """Set up securitas direct."""
+
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=deepcopy(conf)
+        )
+    )
+
```
There's no existing config to import.
```suggestion
```

MartinHjelmare commented on homeassistant/components/securitas_direct/__init__.py:
```diff
@@ -0,0 +1,135 @@
+async def async_setup_entry(hass, config_entry):
+    """Set up securitas direct entry."""
+
+    if config_entry.unique_id is None:
```
There are no existing config entries.

MartinHjelmare reviewed (changes requested):
I started reviewing this PR but it seems you have followed some existing integration and copied over code that is not relevant. Please use our scaffold script when creating a new integration to generate best practice code both for the integration and its tests.

https://developers.home-assistant.io/docs/creating_component_index

When that is done, I'll continue the review.

MartinHjelmare commented:
Please run the scaffold script. When that is applied I'll continue the review. Also please solve the merge conflict.

ludeeus committed:
Generate a seperate UUID for the analytics integration (#48742)

raman325 committed:
Check all endpoints for zwave_js.climate fan mode and operating state (#48800)

* Check all endpoints for zwave_js.climate fan mode and operating state

* fix test

emontnemery committed:
Validate supported_color_modes for MQTT JSON light (#48836)

spacegaier committed:
Fix "notify.events" trim() issue + add initial tests (#48928)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

jjlawren committed:
Check all endpoints for zwave_js.climate hvac_action (#49115)

bdraco committed:
Fix exception in roomba discovery when the device does not respond on the first try (#49360)

currentoor committed:
Mutesync integration (#49679)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Reduce config entry setup/unload boilerplate V-Z (#49789)

tkdrob committed:
Clean up intent_script (#49770)

scop committed:
Huawei LTE unload cleanups (#49788)

bieniu committed:
Add dynamic update interval to Airly integration (#47505)

* Add dynamic update interval

* Update tests

* Improve tests

* Improve comments

* Add MAX_UPDATE_INTERVAL

* Suggested change

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Use async_fire_time_changed to test update interval

* Fix test_update_interval

* Patch dt_util in airly integration

* Cleaning

* Use total_seconds instead of seconds

* Fix update interval test

* Refactor update interval test

* Don't create new context manager

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

dermotduffy committed:
Change motionEye to use a two item device identifier tuple (#49774)

* Change to a two item device identifier tuple.

* Don't use join.

pavoni committed:
Rework roon media player grouping to use media_player base services (#49667)

* Add group/join status attributes to roon player.

* Rework join/unjoin code to use base media_player services.

* Switch join and unjoin to be sync.

frenck committed:
Add Blueprint foundation to Scripts (#48621)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bramkragten committed:
Update frontend to 20210427.0 (#49794)

emontnemery committed:
Fix entity_id change for Google Cast (#49743)

Adminiuga committed:
Don't bind all clusters unconditionally (#49793)

mdz committed:
update python-smarttub dependency to 0.0.24 (#49800)

TheGardenMonkey committed:
Convert DHT to use circuitpython library (#41525)

* Converts DHT to use circuitpython library

* Adds dht requirement to requirements_all

* Cleanup for isort,black and requirements

* Adds validation to prefix with D for PIN

* Sends name over to dhtclient

* Adds exceptions for broad-except

* Adds explict except for raised exception

* Removes unused var

* Moves setting dht var out of try

* Bump to 3.5.8

* Moves non-exception steps to else

* Bump library version

* Bumps version

emontnemery committed:
Correct conversion of RGB and XY colors to RGBW (#49802)

raman325 committed:
Add zwave_js discovery schema for Vision Security ZL7432 (#49510)

* Add zwave_js discovery schema for Vision Security ZL7432

* add tests

* fix test

* add correct fixture

* Make discussed changes

* fix tests

* move event handler

* fix logic to get entity ID

* add test

* make discovery test more explicit

* remove domain from event data

* always provide entity_id key to make automations easier and translate value if possible

* formatting

* comment

* dont overwrite value

bdraco committed:
Bump httpx to 0.18.0 and respx to 0.17.0 (#49805)

bdraco committed:
Fix scrape sensor auth with httpx (#49806)

bdraco committed:
Fix zeroconf test inadvertently matching strings that start with name (#49578)

* Fix zeroconf test inadvertently matching strings that start with name

* adjust

* adjust

tkdrob committed:
Add selectors to icloud services (#49797)

frenck committed:
Small typing tweaks to Light component (#49718)

* Small typing tweaks to Light component

* Use abc collection

tkdrob committed:
Add selector and target to pi_hole services (#49795)

emontnemery committed:
Bump hatasmota to 0.2.11 (#49811)

MartinHjelmare committed:
Fix zwave_js flow set add-on options (#49813)

* Fix zwave_js flow set add-on options

* Improve and deduplicate error messages

* Add more call assertions

* Add final missing call assertion

emontnemery committed:
Improve color conversion for RGBWW lights (#49807)

farmio committed:
Fix KNX unique_id (#49677)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

raman325 committed:
Add additional zwave_js WS API commands for nodes (#49509)

* Add zwave_js WS API commands for node.refresh_values and node.refresh_cc_values

* remove guard since API will be used by frontend

* add missing validation

* use get in case node ID is wrong

* use decorator

* add additional coverage

ludeeus committed:
Don't overwrite existing files when generating configuration (#49815)

bdraco committed:
Serialize websocket messages before placing them in the queue (#49582)

pvizeli committed:
Update base image to version 2021.04.3 (#49817)

firstof9 committed:
Fix white_value causing zwave-js to set non-existing values (#49799)

Kane610 committed:
Last minute tweaks to deCONZ alarm panel entity service(#49819)

Text for entry and exit delay was wrong
Remove functions that need a bit more investigating

ludeeus committed:
Add OS version and board type to analytics (#49725)

Kane610 committed:
Add support for deCONZ alarm events in logbook (#49652)

* Add support for alarm events in logbook

* Update homeassistant/components/deconz/alarm_control_panel.py

Co-authored-by: jjlawren <jjlawren@users.noreply.github.com>

jjlawren committed:
Handle special Plex library sections (#49525)

tkdrob committed:
Add selectors to google services (#49796)

Kane610 committed:
Add tampering sensor to deCONZ integration (#49776)

balloob committed:
Bump frontend to 20210428.0 (#49821)

tkdrob committed:
Add switch platform to goalzero (#48612)

* Add switch platform to goalzero

* fix update interval

* Apply some suggested changes

* pass device class to parent

* Drop passing device_class

* Tweaks

* Drop underscore prefix

bachya committed:
Bump pyairvisual to 5.0.8 (#49823)

chemelli74 committed:
Add service target to Neato (#49803)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

amelchio committed:
Fix color setting in LIFX services (#49822)

raman325 committed:
Set ClimaCell API limit to 500 requests/day (#49828)

frenck committed:
Remove DHT from Raspberry Pi machine builds (#49829)

frenck committed:
Bump version to 2021.6.0dev0 (#49830)

homeassistant committed:
[ci skip] Translation update

mkeesey committed:
Remove references to hass.data in harmony tests (#49836)

Instead, just use the mocks directly.

KapJI committed:
Validate if modules in mypy config exist (#49810)

KapJI committed:
Relax type annotation for DataUpdateCoordinator data (#49827)

fabaff committed:
Upgrade py-cpuinfo to 8.0.0 (#49833)

emontnemery committed:
Add color_mode to demo light (#49694)

* Update demo light with color mode support

* Add rgbw and rgbww color properties

* Update demo light

* Tweak

* Remove unneeded _clear_colors

ludeeus committed:
Add onboarded key to analytics WS command (#49751)

frenck committed:
hassfest detect built-in domain override for custom integrations (#49845)

frenck committed:
Upgrade black to 21.4b2 (#49841)

KapJI committed:
Define AddEntitiesCallback type (#49812)

frenck committed:
Remove example entry from PR template (#49842)

KapJI committed:
Rename FlowResultDict to FlowResult (#49847)

raman325 committed:
Improve Waze Travel Time import and naming logic (#49838)

KapJI committed:
Disable strict type checks for tests (#49851)

janiversen committed:
Catch missing/unavailable response from modbus (#49632)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bieniu committed:
Fix `host_valid()` logic in BraviaTV config flow (#49857)

TheGardenMonkey committed:
Add codeowners for DHT (#49855)

djtimca committed:
Add Omnilogic Switch platform (#42116)

* Bump omnilogic dependency to 0.4.4 to fix Lights key error.

* Bumped dependency to 0.4.5.

* Fixed pump type issue for unique pool structure.

* Create full platform bundle for final testing and PR to Home Assistant dev.

* Removed logger instances not required.

* Fixed lint issues.

* Fixed pylint issues.

* Fix pylint issues. Fix issue with pH sensor offset.

* Stripped light, water_heater platform for PR submit.

* Correct pH and ORP sensor report to unknown with offset if pump is off.

* Moving guard condition check to helper function.

* Update to asyncio.sleep to wait for switch status delay in Hayward API status.

* Removed sleep, added state delay to handle slow Hayward API state update response.

* Fix flake8 issue.

* Fix flake8 issue.

* Fix isort issue.

* Addressed PR Comments.

* Addressed PR comments. Corrected Unit of Measure for sensor where pump speed is not variable.

* Fix pylint issue.

* Address pylint issue.

* Update homeassistant/components/omnilogic/switch.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

bieniu committed:
Improve type annotations for the Brother integration (#49771)

bieniu committed:
Fix `host_valid()` logic in DuneHD config flow (#49860)

raman325 committed:
Use EVENT_HOMEASSISTANT_STARTED instead of EVENT_HOMEASSISTANT_START (#49861)

raman325 committed:
Improve Google Travel Time import and naming logic (#49839)

KapJI committed:
Rename FlowResultDict in scaffold template (#49858)

bieniu committed:
Fix `host_valid()` logic in Vilfo config flow (#49862)

janiversen committed:
Complete modbus device response tests (#49633)

* Prepare test harness for new pymodbus return types.

Use pytest.fixture to mock pymodbus.
Use pytest.fixture to load modbus using mocked pymodbus
Add test of Exception/IllegalResponse/ExceptionResponse from pymodbus.

* Modbus.py is back at 100% test coverage.

* Added assert mock.called.

* add mock reset.

thecode committed:
Add color modes to Shelly light (#49867)

balloob committed:
Add auto_off to binary sensor template entity (#49615)

AaronDavidSchneider committed:
Fix Fritz device tracker multiple routers (#49808)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

scop committed:
Restore missing Huawei LTE cleanup on HA stop (#49871)

Refs https://github.com/home-assistant/core/pull/49788#discussion_r623071013

raman325 committed:
Fix options flow bugs in Google/Waze Time Travel (#49866)

karliemeads committed:
Apply default light profile only when light is toggled from off to on (#49376)

Co-authored-by: Franck Nijhof <git@frenck.dev>

jjlawren committed:
Add additional Sonos codeowner (#49879)

frenck committed:
Cleanup of unused connection_class logic (#49865)

Co-authored-by: Joakim Sørensen <joasoe@gmail.com>

frenck committed:
Fix jumpy Verisure Wallplug state (#49880)

bramkragten committed:
Update frontend to 20210407.1 (#49882)

homeassistant committed:
[ci skip] Translation update

bachya committed:
Remove no-longer-functioning SimpliSafe websocket support (#49876)

* Remove no-longer-functioning SimpliSafe websocket support

* Linting

jjlawren committed:
Do not pass hass.data to Sonos entities (#49881)

currentoor committed:
Reduced polling interval for mutesync integration (#49884)

raman325 committed:
Don't validate inputs on platform setup for Google/Waze travel time (#49886)

* Don't validate inputs on platform setup for Google/Waze travel time

* feedback from other PR

scop committed:
Allow str 1..n-tuples in device registry identifier type hints (#49864)

This reflects current practice, but the intent has been to have them as
2-tuples, and a future change is likely to start enforcing that (again).

Refs https://github.com/home-assistant/core/pull/49670#discussion_r622344872

frenck committed:
Update frontend to 20210429.0 (#49896)

frenck committed:
Fix unexpected data in Google Travel Time config flow import (#49903)

frenck committed:
Fix unexpected data in Waze config flow import (#49902)

emontnemery committed:
Add color_mode support to zwave_js light (#49588)

dmulcahey committed:
Small ZHA code cleanup (#49908)

frenck committed:
Upgrade pytest-xdist to 2.2.1 (#49904)

bieniu committed:
Split long string in DuneHD test (#49900)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

raman325 committed:
Add device specific discovery for Heatit Z-TRM3 thermostat (#49804)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bieniu committed:
Remove unused code from Brother config flow (#49889)

janiversen committed:
Restructure modbus switch, and solve read/write conflicts (#49386)

Remove old style configuration from switch.
- The old style configuration allowed a number of illegal
  configurations, instead of adding if...log. in setup we
  only support the new configuration style.

Add new/changed configuration switch.

Removed verify_state and cleaned configuration to avoid possible
illegal configurations.

dmulcahey committed:
Move ZHA config panel section translations to the backend (#49816)

KapJI committed:
Use AddEntitiesCallback type, pt.2 (#49921)

frenck committed:
Enable strict typing for the Elgato integration (#49920)

KapJI committed:
Make CoordinatorEntity generic (#49854)

bramkragten committed:
Update frontend to 20210430.0 (#49928)

hunterjm committed:
Bump xbox-webapi to 2.0.11 (#49929)

janiversen committed:
Start Modbus cover timer independent if there is a state to restore (#49919)

* Start timer independent if there is a state to restore.

* Update test cases, that now start working.

balloob committed:
Iqvia to check bad fetch during setup (#49931)

jjlawren committed:
Use header URI in Plex config flow (#49915)

scop committed:
Make device info a TypedDict (#49670)

frenck committed:
Clean up config entry handlers decorator from config flows (#49932)

bdraco committed:
Reduce homekit executor calls at start (#49925)

- We do not need to load/persist since we do it in setup

scop committed:
Use device registry CONNECTION_* constants (#49923)

bdraco committed:
Avoid writing esphome data if its already up to date on disk (#49933)

homeassistant committed:
[ci skip] Translation update

dermotduffy committed:
Repair elgato types (#49936)

frenck committed:
Clean up connection classes in integrations A-C (#49888)

frenck committed:
Clean up connection classes in integrations D-G (#49890)

frenck committed:
Clean up connection classes in integrations H-L (#49891)

frenck committed:
Clean up connection classes in integrations M-O (#49892)

frenck committed:
Clean up connection classes in integrations T-Z (#49894)

frenck committed:
Clean up connection classes in integrations P-S (#49893)

bdraco committed:
Remove unused imports in fritz, nest, and somfy to fix CI (#49940)

Drafteed committed:
Update LG Netcast to use new backend library (#49927)

* Update LG Netcast to use new backend library

Bump lgnetcast to 0.3.3

* Add codeowner in LG Netcast

thecode committed:
Fix light services descriptions (#49951)

janiversen committed:
Please mypy (axis). (#49949)

* Please mypy (axis).

* Update homeassistant/components/axis/config_flow.py

Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>

Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>

thecode committed:
Shelly light color mode bugfix (#49948)

janiversen committed:
Fix ihc typing (#49946)

Adminiuga committed:
Bump up ZHA dependencies (#49959)

bdraco committed:
Remove YAML support from sense (#49935)

bachya committed:
Bump simplisafe-python to 9.6.10 (#49962)

janiversen committed:
Catch non payload modbus messages (#49910)

frenck committed:
Remove ServiceCallType alias from codebase (#49844)

bdraco committed:
Bump pysonos to 0.0.44 to fix client session race (#49964)

Fixes #49954

scop committed:
Improve device registry internal typing (#49924)

milanmeu committed:
Replace dict with DeviceInfo (#49950)

* Replace dict with DeviceInfo

* Clean up

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bachya committed:
Fix KeyError in IQVIA (#49968)

homeassistant committed:
[ci skip] Translation update

alandtse committed:
Bump Tesla dependency teslajsonpy to 0.18.3 (#49939)

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
Handle different entity_id formats (#49969)

janiversen committed:
Enable homeassistant.update_entity service for all modbus platforms (#49918)

* Rename _update() to update()

A platform neeed a function update(), even though
polling is false, this is due to the service:
homeassistant.update_entity, which calls update()

* Update test harnesss to script testing.

Test homeassistant.update_entity in all platforms.
This call calls update() in the platform to get
a new reading.

* Add reuse parameter.

* Move service call from helper to tests.

* Change run_service_update --> prepare_service_update.

* Remove entity_id parameter.

dgomes committed:
Improves UX of Utility Meter services (#48556)

bdraco committed:
Add zeroconf discovery to rachio (#49973)

frenck committed:
Upgrade elgato to 2.1.0 (#49975)

TheZoker committed:
Upgrade yeelight to 0.6.2 (#49995)

bdraco committed:
Add dhcp discovery to tado (#49992)

milanmeu committed:
Abstract Rituals API data processing to PyPI (#49872)

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Check exception causes for matching strings during recorder migration (#49999)

thecode committed:
Fix Shelly external sensors invalid 999 value (#49994)

bramkragten committed:
Correct the selector for frontend.set_theme service (#49952)

balloob committed:
Make hassfest service validation faster (#50003)

bdraco committed:
Add reauth support to myq (#49998)

bramkragten committed:
Fix saving a scene (#49980)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bdraco committed:
Defer writing http config until after startup has calmed down (#50000)

thecode committed:
Fix Shelly battery operated devices value rounding (#49966)

helto4real committed:
Update smhi package to 1.0.15 (#49987)

unaiur committed:
Upgrade maxcube-api dependency to 0.4.3 (#49982)

This new version only contains improvements in logs
to try to debug issue #49482

dermotduffy committed:
Fix types for shell command (#50004)

farmio committed:
Fix KNX light unique_id (#49967)

* migrate light unique_id

* review changes

farmio committed:
Add color modes to KNX light (#49883)

* color_modes support

* return brightness for color lights

even when no brightness address is set

* apply brightness for color lights

* remove unneeded constants

austinmroczek committed:
Add unique_id to TotalConnect alarm_control_panel (#49961)

* add unique_id to alarm_control_panel

* Update homeassistant/components/totalconnect/alarm_control_panel.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Add dhcp discovery to hunterdouglas_powerview (#49993)

* Add dhcp discovery to hunterdouglas_powerview

* avoid dupe flow

* cleanup

* cleanup

chemelli74 committed:
Title and internal descriptions updates for Fritzbox (#49047)

frenck committed:
Add typing to async_register_entity_service (#50015)

ludeeus committed:
Handle Timeout exceptions in system_health (#50017)

frenck committed:
Fix Blink entity service schema (#50019)

bieniu committed:
Improve type annotations in Brother integration (#49989)

frenck committed:
Add small async_get_current_platform helper method (#50014)

Mariusthvdb committed:
Mitigate NMBS key errors (#50026)

on Liveboard and connections as documented in issue #48824

frenck committed:
Add Identify service to Elgato integration (#49990)

bramkragten committed:
Update frontend to 20210503.0 (#50036)

frenck committed:
Migrate to async_get_current_platform everywhere (#50034)

frenck committed:
Fix ELKM1 entity service schema (#50020)

frenck committed:
Fix Nexia entity service schema (#50027)

frenck committed:
Fix Harmony entity service schema (#50025)

frenck committed:
Fix Genius Hub entity service schema (#50024)

frenck committed:
Restore dictionary constants in Elgato device info (#50013)

KapJI committed:
Create separate entries for each component in mypy.ini (#50030)

bsmappee committed:
Bump pysmappee to 0.2.25 (#50031)

bdraco committed:
Add reauth support to flume (#49991)

Danielhiversen committed:
Upgrade Tibber library, new grid prices for Glitre Energi (#50029)

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

yannickglt commented:
Would be awesome to get this pull request merged. Thanks for this amazing work @nragon!

janiversen committed:
Fix modbus typing (#49938)

Add changes needed to please mypy and follow the coding rules
of the project.

homeassistant committed:
[ci skip] Translation update

chemelli74 committed:
Create Fritz device and connectivity sensor (#49699)

Co-authored-by: J. Nick Koston <nick@koston.org>

dermotduffy committed:
Enable mypy for motionEye (aye aye!) (#49738)

nielstron committed:
Add SyncThru binary sensors (#48114)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Handle missing transport_state on media update in sonos (#50051)

Kane610 committed:
Use last_step marker on UniFi options flow (#50053)

dependabot[bot] committed:
Bump codecov/codecov-action from v1.4.1 to v1.5.0 (#50061)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.4.1 to v1.5.0.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.4.1...a1ed4b322b4b38cb846afb5a0ebfa17086917d27)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

RobBie1221 committed:
Add configuration flow for Buienradar integration (#37796)

* Add configuration flow for Buienradar integration

* Update buienradar camera tests to work with config flow

* Update buienradar weather tests to work with config flow

* Update buienradar sensor tests to work with config flow

* Remove buienradar config_flow tests to pass tests

* Add config flow tests for buienradar integration

* Increase test coverage for buienradar config_flow tests

* Move data into domain

* Remove forecast option

* Move data to options

* Remove options from config flow

* Adjust tests

* Adjust string

* Fix pylint issues

* Rework review comments

* Handle import

* Change config flow to setup camera or weather

* Fix tests

* Remove translated file

* Fix pylint

* Fix flake8

* Fix unload

* Minor name changes

* Update homeassistant/components/buienradar/config_flow.py

Co-authored-by: Ties de Kock <ties@tiesdekock.nl>

* Remove asynctest

* Add translation

* Disable sensors by default

* Remove integration name from translations

* Remove import method

* Drop  selection between platforms, disable camera by default

* Minor fix in configured_instances

* Bugfix in weather

* Rework import

* Change unique ids of camera

* Fix in import

* Fix camera tests

* Fix sensor test

* Fix sensor test 2

* Fix config flow tests

* Add option flow

* Add tests for option flow

* Add import tests

* Some cleanups

* Apply suggestions from code review

Apply code suggestions

Co-authored-by: Franck Nijhof <git@frenck.dev>

* Fix isort,black,mypy

* Small tweaks and added typing to new parts

* Fix review comments (1)

* Apply suggestions from code review

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Fix review comments (2)

* Fix issues

* Fix unique id

* Improve tests

* Extend tests

* Fix issue with unload

* Address review comments

* Add warning when loading platform

* Add load/unload test

Co-authored-by: Ties de Kock <ties@tiesdekock.nl>
Co-authored-by: Franck Nijhof <git@frenck.dev>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

milanmeu committed:
Fix and enable type checks in Rituals Perfume Genie (#49947)

KapJI committed:
Use AddEntitiesCallback type, pt.4 (#49955)

thecode committed:
Catch Shelly set state exceptions when device is inaccessible (#50064)

frenck committed:
Clean up stale config schema from deCONZ (#50081)

frenck committed:
Finalize clean up connection classes (#49895)

frenck committed:
Upgrade pytest to 6.2.4 (#50077)

frenck committed:
Remove YAML configuration from DoorBird (#50082)

puddly committed:
Bump zigpy-znp from 0.4.0 to 0.5.1 (#50086)

* Bump zigpy-znp from 0.4.0 to 0.5.0

* Add zigpy-znp to ZHA debug logging config

* Bump zigpy-znp from 0.5.0 to 0.5.1

frenck committed:
Deprecate speedtest.net YAML config (#50072)

frenck committed:
Remove stale config schema from ESPHome integration (#50083)

balloob committed:
Guard logbook assuming entity ID is a string (#50047)

frenck committed:
Remove YAML configuration from Daikin (#50080)

frenck committed:
Remove deprecated LIFX Legacy integration (#50069)

frenck committed:
Deprecate Abode YAML configuration (#50075)

bramkragten committed:
Fix updating owner user/auth (#50087)

Check if `is_active` is in update msg

KapJI committed:
Use AddEntitiesCallback type, pt.3 (#49953)

farmio committed:
Fix KNX climate unque_id (#50054)

bramkragten committed:
Update frontend to 20210504.0 (#50093)

janiversen committed:
Cleanup modbus binary_sensor signature (#49809)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

homeassistant committed:
[ci skip] Translation update

frenck committed:
Upgrade yamllint to 1.26.1 (#50060)

frenck committed:
Clean up deprecation message & config schema from Cloudflare (#50079)

dermotduffy committed:
Bump motioneye-client to v0.3.6 . (#50096)

frenck committed:
Clean up pylint comments (#49334)

RobBie1221 committed:
Improve buienradar tests (#50101)

frenck committed:
Upgrade black to 21.5b0 (#50102)

frenck committed:
Upgrade luftdaten to 0.6.5 (#50103)

vlebourl committed:
Fix fitbit RuntimeError: I/O must be done in the executor (#50058)

frenck committed:
Upgrade pyupgrade to v2.14.0 (#50059)

frenck committed:
Remove YAML configuration from Verisure (#50076)

frenck committed:
Deprecate Glances YAML configuration (#50085)

frenck committed:
Deprecate Denon HEOS YAML configuration (#50104)

frenck committed:
Deprecate Freebox YAML configuration (#50084)

emontnemery committed:
Add color_mode support to tasmota light (#49599)

ludeeus committed:
Allign error handling for restart for hassio with core (#50114)

* Allign error handling for restart for hassio with core

* Reuse HASS_DOMAIN

* Address comments

benleb committed:
Remove surepetcare usage of deprecated config options (#50113)

emontnemery committed:
Fix Tasmota color scaling and RGBW lights (#50120)

epenet committed:
Drop OWFS support in onewire (#50121)

timmo001 committed:
System Bridge Integration (#48156)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Add support for Elgato Light Strip (#49988)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

homeassistant committed:
[ci skip] Translation update

rytilahti committed:
Bump python-miio dependency (#50129)

bdraco committed:
Bump sqlalchemy to 1.4.13 (#50138)

bdraco committed:
Bump zeroconf to 0.30.0 to fix thread safety race (#50130)

pvizeli committed:
Adjust GRPC wheel build (#50119)

tkdrob committed:
Fix group selector (#50088)

emontnemery committed:
Improve translation strings for MQTT config and option flows (#50018)

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Use shared httpx client in gogogate2 (#45575)

frenck committed:
Strictly type Twente Milieu integration (#50062)

frenck committed:
Deprecate Canary YAML configuration (#50078)

frenck committed:
Allow passing options in config flow entry creation (#49912)

MartinHjelmare committed:
Fix zwave_js websocket api KeyError on unloaded entry (#50154)

bdraco committed:
Add rainmachine discovery (#49970)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

cgtobi committed:
Bump PyRMVtransport to 0.2.3 (#50183)

Xeovar commented:
Wow, what happened? 😵

nragon commented:
Following up on new clean PR. My bad.

alnavasa commented:
> Following up on new clean PR. My bad.

can you link the new PR? to follow it. @nragon

nragon commented:
#50233
